counterpartylib/lib/messages/issuance.py

```diff
@@ -246,37 +249,65 @@ def compose (db, source, transfer_destination, asset, quantity, divisible, descr
 
     call_date, call_price, problems, fee, description, divisible, reissuance, reissued_asset_longname = validate(db, source, transfer_destination, asset, quantity, divisible, callable_, call_date, call_price, description, subasset_parent, subasset_longname, util.CURRENT_BLOCK_INDEX)
     if problems: raise exceptions.ComposeError(problems)
 
     asset_id = util.generate_asset_id(asset, util.CURRENT_BLOCK_INDEX)
-    data = struct.pack(config.TXTYPE_FORMAT, ID)
-    if len(description) <= 42:
-        curr_format = FORMAT_2 + '{}p'.format(len(description) + 1)
+    if subasset_longname is None or reissuance:
+        # Type 20 standard issuance FORMAT_2 >QQ??If
+        #   used for standard issuances and all reissuances
+        data = struct.pack(config.TXTYPE_FORMAT, ID)
+        if len(description) <= 42:
+            curr_format = FORMAT_2 + '{}p'.format(len(description) + 1)
+        else:
+            curr_format = FORMAT_2 + '{}s'.format(len(description))
+        data += struct.pack(curr_format, asset_id, quantity, 1 if divisible else 0, 1 if callable_ else 0,
+            call_date or 0, call_price or 0.0, description.encode('utf-8'))
     else:
-        curr_format = FORMAT_2 + '{}s'.format(len(description))
-    data += struct.pack(curr_format, asset_id, quantity, 1 if divisible else 0, 1 if callable_ else 0,
-        call_date or 0, call_price or 0.0, description.encode('utf-8'))
+        # Type 21 subasset issuance SUBASSET_FORMAT >QQ?B
+        #   Used only for initial subasset issuance
+        # compacts a subasset name to save space
+        compacted_subasset_longname = util.compact_subasset_longname(subasset_longname)
+        compacted_subasset_length = len(compacted_subasset_longname)
+        data = struct.pack(config.TXTYPE_FORMAT, SUBASSET_ID)
+        curr_format = SUBASSET_FORMAT + '{}s'.format(compacted_subasset_length) + '{}s'.format(len(description))
+        data += struct.pack(curr_format, asset_id, quantity, 1 if divisible else 0, compacted_subasset_length, compacted_subasset_longname, description.encode('utf-8'))
+
     if transfer_destination:
         destination_outputs = [(transfer_destination, None)]
     else:
         destination_outputs = []
     return (source, destination_outputs, data)
 
-def parse (db, tx, message):
+def parse (db, tx, message, message_type_id):
     issuance_parse_cursor = db.cursor()
 
     # Unpack message.
     try:
-        if (tx['block_index'] > 283271 or config.TESTNET) and len(message) >= LENGTH_2: # Protocol change.
+        subasset_longname = None
+        if message_type_id == SUBASSET_ID and util.enabled('subassets', block_index=tx['block_index']):
+            # parse a subasset original issuance message
+            asset_id, quantity, divisible, compacted_subasset_length = struct.unpack(SUBASSET_FORMAT, message[0:SUBASSET_FORMAT_LENGTH])
+            description_length = len(message) - SUBASSET_FORMAT_LENGTH - compacted_subasset_length
```

Comment:
Good catch.  A length of zero was getting through this check and issuing a numeric asset.

I added an explicit validation call in the parse function to prevent this.

I also added unit tests to cover this situation.